### PR TITLE
Fixed output derivation for wheels fetching

### PR DESCRIPTION
**Pull request by RaitoBezarius**
Due to the way PyPI routes its wheels (security reasons apparently), it's impossible to guess the URL, without calling the PyPI API.

For posterity, why `/${kind}/` cannot work is due to this: https://gist.github.com/di/93019b51582899c294cfd76ee2d0a4a3 (list of all python versions in warehouse).

The way this PR attempts to solve it is to call the API and just replace `fetchurl` which is anyway also a fixed-output derivation.

### CODE REVIEW

File: lib.nix [hunk 1]
```diff
@@ -82,6 +82,38 @@ let
...
+        buildInputs = with pkgs; [ curl jq ];
+
+        phases = ["buildPhase"];
+        preferLocalBuild = true;
+        impureEnvVars = [
+          "http_proxy" "https_proxy" "ftp_proxy" "all_proxy" "no_proxy"
+        ];
+
+        buildPhase = ''
+          curl -k $(curl -k https://pypi.org/pypi/${pname}/json | jq -r '.releases."${version}"[] | select(.filename == "${file}") | .url') --output $out
```
Comment (adisbladis) [inline lib.nix:113]: Could we try the calculated url and only then fall back to round-tripping to the API?

Comment (RaitoBezarius) [reply lib.nix]: @adisbladis I can do this

Comment (RaitoBezarius) [reply lib.nix]: Done

**Comment by RaitoBezarius** [comment]
Wohoo ! :tada:
@adisbladis 
What I wrote in the overrides though is a bit hacky, don't you think? Basically, we should prevent standard overrides to mess up with wheel preference (in general)?

**Pull request merged**